xla/service/cpu/tests/onednn_convolution_test.cc

```diff
@@ -32,159 +33,221 @@ limitations under the License.
+    std::ostringstream stream;
+    std::for_each(
+        fused_ops.begin(), fused_ops.end(),
+        [&](const absl::string_view& arg) { stream << "\"" << arg << "\","; });
+    std::string fusions = stream.str();
+    if (fused_ops.size() > 0) {
+      fusions.pop_back();
+      return absl::Substitute(
+          conv_rewrite_str_,
+          absl::Substitute(conv_rewrite_fusions_str_, fusions),
+          GetOptimizationsString());
+    }
+    return absl::Substitute(conv_rewrite_str_, "", GetOptimizationsString());
   }
 
-  const char* convolution_module_str = R"(
-  HloModule convolution.test.bf16
+  // TODO(intel-tf): Remove this and simplify patterns when Elemental BF16 is
+  // fully supported.
+  PrimitiveType PromotedDtype() {
+    // BF16 is promoted to F32 because not all HLO Instructions currently
+    // support BF16 computations. Meanwhile, FP32 and FP16 elementwise
+    // instructions are not promoted and remain unchanged.
+    switch (dtype) {
+      case F16:
+        return F16;
+      case F32:
+      case BF16:
+      default:
+        return F32;
+    }
+  }
 
-  ENTRY convolution.test.bf16 {
-    p0 = bf16[8,4,5,5,1] parameter(0)
-    p1 = bf16[3,3,3,1,32] parameter(1)
-    ROOT conv = bf16[8,4,5,5,32] convolution(p0, p1), window={size=3x3x3 pad=1_1x1_1x1_1}, dim_labels=b012f_012io->b012f
-})";
+  void AdjustToleranceForDtype(PrimitiveType for_type, float atol_arg,
+                               float rtol_arg) {
+    if (dtype == for_type) {
+      atol = atol_arg;
+      rtol = rtol_arg;
+    }
+  }
 
-  EXPECT_TRUE(RunAndCompare(convolution_module_str, ErrorSpec{1e-4, 1e-4}));
-  MatchOptimizedHlo(convolution_module_str, conv_rewrite_str_);
-}
+  std::string PromotedDtypeToString() {
+    auto promoted_type = PromotedDtype();
+    return primitive_util::LowercasePrimitiveTypeName(promoted_type);
+  }
 
-TEST_F(ConvolutionTest, Simple2DTestF16) {
-  if (!IsSupportedType(PrimitiveType::F16)) {
-    GTEST_SKIP() << "CPU does not support F16.";
+  void RunCompareAndMatchOptimizedHlo(
+      const absl::string_view outline,
+      const std::vector<absl::string_view> fused_ops) {
+    const std::string convolution_module_str =
+        absl::Substitute(outline, dtypeString, PromotedDtypeToString());
+    EXPECT_TRUE(RunAndCompare(convolution_module_str, ErrorSpec{atol, rtol}));
+    MatchOptimizedHlo(convolution_module_str,
+                      ConvStringWithOptimizations(fused_ops));
   }
+};
 
-  const char* convolution_module_str = R"(
-  HloModule convolution.test.f16
+TEST_P(ConvolutionTest, Simple2DTest1) {
+  const absl::string_view outline = R"(
+  HloModule convolution.test
+
+  ENTRY convolution.test {
+    arg.0 = $0[1,22,22,1] parameter(0)
+    reshape.0 = $0[1,22,22,1] reshape(arg.0)
+    arg.1 = $0[8,8,1,1] parameter(1)
+    reshape.1 = $0[8,8,1,1] reshape(arg.1)
+    convolution.0 = $0[1,11,11,1] convolution(reshape.0, reshape.1), window={size=8x8 stride=2x2 pad=3_3x3_3}, dim_labels=b01f_01io->b01f
```

Comment:
Please wrap any lines longer than 80 characters. Also, I think we can drop `dim_labels`.
```suggestion
    convolution.0 = $0[1,11,11,1] convolution(reshape.0, reshape.1), 
                        window={size=8x8 stride=2x2 pad=3_3x3_3}
```

Reply:
Wrapped all long lines. The parser throws the following error if I remove `dim_labels`
`error: attribute dim_labels is expected but not seen` so, I did not remove `dim_labels`